### PR TITLE
新增迷宮模式(Add new mazes) 

### PR DESCRIPTION
**Pull request by YiKaiWu1125**
+ 目前為簡易版UI (未來可改進)
+ 提供兩關卡可測試(easy,hard)
+ 解決 #4 Add new mazes

### CODE REVIEW

File: real_mazes/maze_game.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            if cv2.waitKey(5) & 0xFF == 27:
+                break
+            if sta == 'game_over':
+                time.sleep(5)
+                break
+    cap.release()
+    cv2.destroyAllWindows()                 
+
+if __name__ == '__main__':
+    camera_id = 1#'test.mp4'
```
Comment (asas1asas200) [inline real_mazes/maze_game.py]: Consider changing to `0` for default camera ID, or pass it by command line arguments.

File: real_mazes/maze_game.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        #cv2.putText(image, "play again", (160, 320),
+        #            cv2.FONT_HERSHEY_PLAIN, 3, (0, 25, 240), 3)  # 再玩一次
+        #cv2.rectangle(image, (155, 350), (420, 400),
+        #              (0, 25, 240), 5)   # 遊戲結束
+        #cv2.putText(image, "end game", (160, 390),
+        #            cv2.FONT_HERSHEY_PLAIN, 3, (0, 25, 240), 3)  # 遊戲結束
+    #print("sta is :"+str(sta))
+    return img,sta,game_begin_time,game_end_time
+
+def begin_maze_gmae(camera_id,picture_name,width,hight):
```
Comment (asas1asas200) [inline real_mazes/maze_game.py:98]: Typo `begin_maze_game`.

**Review by asas1asas200 — CHANGES_REQUESTED** [review]
There are two problems need to be solved:
+ Are you tested it on your machine, it cannot run on my machine:
  ```shell
  $ python maze_game.py
  video init Image Size: 640 x 480
    video fix Image Size: 1280 x 720
  INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
  Traceback (most recent call last):
    File "/home/zeng/Desktop/ElecTrap_backend/real_mazes/maze_game.py", line 168, in <module>
      begin_maze_gmae(camera_id,picture_name,width,hight)
    File "/home/zeng/Desktop/ElecTrap_backend/real_mazes/maze_game.py", line 150, in begin_maze_gmae
      img,sta,game_begin_time,game_end_time = maze_judgment_status(img,unit8_img,binarization_arr,x,y,sta,game_begin_time,game_end_time)
    File "/home/zeng/Desktop/ElecTrap_backend/real_mazes/maze_game.py", line 59, in maze_judgment_status
      img = draw_maze(img,unit8_img)
    File "/home/zeng/Desktop/ElecTrap_backend/real_mazes/maze_game.py", line 44, in draw_maze
      image = cv2.add(image,unit8_img)
  cv2.error: OpenCV(4.6.0) /io/opencv/modules/core/src/arithm.cpp:647: error: (-209:Sizes of input arguments do not match) The operation is neither 'array op array' (where arrays have the same size and the same number of channels), nor 'array op scalar', nor 'scalar op array' in function 'arithm_op'
  ```
  If this code work for you, show off your python version and pip packages version.
+ Is `easy.png` and `hard.png` the same file?

**Comment by asas1asas200** [comment]
Consider rewriting it as class for [OOP](https://en.wikipedia.org/wiki/Object-oriented_programming), this is Python not C/C++.

**Comment by YiKaiWu1125** [comment]
Are you tested it on your machine, it cannot run on my machine
`回復: 我的所有電腦上皆可順利執行，至於您無法順離執行原因，應該是您的鏡頭好像只支持到1280 x 720，無法支援到1920*1080 ，而目前本地圖僅支持1920*1080，後續將會更新鏡頭解析度相容性問題`
If this code work for you, show off your python version and pip packages version.
`回復:`
```
absl-py==1.2.0
attrs==21.4.0
click==8.1.3
colorama==0.4.5
cycler==0.11.0
Flask==2.1.2
fonttools==4.34.4
importlib-metadata==4.12.0
itsdangerous==2.1.2
Jinja2==3.1.2
kiwisolver==1.4.4
MarkupSafe==2.1.1
matplotlib==3.5.2
mediapipe==0.8.10.1
numpy==1.21.6
opencv-contrib-python==4.6.0.66
opencv-python==4.6.0.66
packaging==21.3
Pillow==9.2.0
pip==22.2
protobuf==3.20.1
pyparsing==3.0.9
python-dateutil==2.8.2
setuptools==47.1.0
six==1.16.0
typing_extensions==4.3.0
Werkzeug==2.2.0
zipp==3.8.1
```
Is easy.png and hard.png the same file?
`回復:當然不一樣，hard.png的地圖真的超難喔`

**Comment by asas1asas200** [comment]
> Are you tested it on your machine, it cannot run on my machine `回復: 我的所有電腦上皆可順利執行，至於您無法順離執行原因，應該是您的鏡頭好像只支持到1280 x 720，無法支援到1920*1080 ，而目前本地圖僅支持1920*1080，後續將會更新鏡頭解析度相容性問題` If this code work for you, show off your python version and pip packages version. `回復:`
> 
> ```
> absl-py==1.2.0
> attrs==21.4.0
> click==8.1.3
> colorama==0.4.5
> cycler==0.11.0
> Flask==2.1.2
> fonttools==4.34.4
> importlib-metadata==4.12.0
> itsdangerous==2.1.2
> Jinja2==3.1.2
> kiwisolver==1.4.4
> MarkupSafe==2.1.1
> matplotlib==3.5.2
> mediapipe==0.8.10.1
> numpy==1.21.6
> opencv-contrib-python==4.6.0.66
> opencv-python==4.6.0.66
> packaging==21.3
> Pillow==9.2.0
> pip==22.2
> protobuf==3.20.1
> pyparsing==3.0.9
> python-dateutil==2.8.2
> setuptools==47.1.0
> six==1.16.0
> typing_extensions==4.3.0
> Werkzeug==2.2.0
> zipp==3.8.1
> ```
> 
> Is easy.png and hard.png the same file? `回復:當然不一樣，hard.png的地圖真的超難喔`

But these look too similar:

**Comment by asas1asas200** [comment]
Use `Quote reply` to reply the context next time.

**Comment by asas1asas200** [comment]
And it seems low fps on my machine, what about yours?

File: real_mazes/maze_game.py (hunk 1 above)
Comment (YiKaiWu1125) [reply real_mazes/maze_game.py]: 已先改為
`camera_id = 0`

File: real_mazes/maze_game.py (hunk 2 above)
Comment (YiKaiWu1125) [reply real_mazes/maze_game.py:98]: 已更正為 `begin_maze_game`

**Comment by YiKaiWu1125** [comment]
> And it seems low fps on my machine, what about yours?

我的跑起來還可以 目測有大於15fps (還是會需要先嵌入計算fps的畫面)

**Comment by YiKaiWu1125** [comment]
> Use `Quote reply` to reply the context next time. 

ok 好的~~

**Comment by asas1asas200** [comment]
> > And it seems low fps on my machine, what about yours?
> 
> 我的跑起來還可以 目測有大於15fps (還是會需要先嵌入計算fps的畫面)

Nice to have.

This PR will be merged after doing this.

**Comment by YiKaiWu1125** [comment]
> > > And it seems low fps on my machine, what about yours?
> > 
> > 
> > 我的跑起來還可以 目測有大於15fps (還是會需要先嵌入計算fps的畫面)
> 
> Nice to have.
> 
> This PR will be merged after doing this.

已新增fps資訊 在畫面中

**Pull request merged**